src/disco/keyguard/fd_keyguard_match.c

```diff
@@ -166,6 +172,9 @@ fd_keyguard_payload_matches_gossip_msg( uchar const * data,
      location). */
   if( sz<36UL ) return 0;
 
+  /* Pings and pongs are prefixed with a string */
+  if ( fd_keyguard_payload_matches_ping_msg( data, sz) ) return 1;
```

Comment:
Can you work with richie to verify that the umabiguity proof still runs and says this change is valid?

Reply:
I'm not convinced yet that this is unambiguous, you need to run to prover to show that no other message type could have these `SOLANA_PING_PONG` bytes in the beginning.